### PR TITLE
III-4815 ignore prices with category uitpas

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- tests for update PriceInfo with UiTPAS

### Changed

- Ignore UiTPAS Prices in `PriceInfoDenormalizer`
- Include check onUiTPAS Prices in updatePriceInfo of Aggregate

---
Ticket: https://jira.uitdatabank.be/browse/III-4815

### CODE REVIEW

File: tests/Event/EventTest.php [hunk 1]
```diff
@@ -535,6 +535,267 @@ function (Event $event) use ($bookingInfo) {
             );
     }
 
+    /**
+     * @test
+     */
+    public function it_handles_keeps_uitpas_prices(): void
```
Comment (bertramakers) [inline tests/Event/EventTest.php:541]:
```suggestion
    public function it_keeps_existing_uitpas_prices_on_price_info_update(): void
```

File: tests/Event/EventTest.php [hunk 2]
```diff
@@ -535,6 +535,267 @@ function (Event $event) use ($bookingInfo) {
...
+                    )
+                )
+            )
+            ->then([]);
+    }
+
+    /**
+     * @test
+     */
+    public function it_ignores_an_update_with_equal_prices_with_uitpas_on_uitpas_prices(): void
```
Comment (bertramakers) [inline tests/Event/EventTest.php:739]: I'm not sure what the `_with_uitpas_on_uitpas_prices` means. Copy/paste mistake from the previous test?

File: tests/Event/EventTest.php [hunk 3]
```diff
@@ -535,6 +535,267 @@ function (Event $event) use ($bookingInfo) {
...
+                            ),
+                        ])
+                ),
+            ]);
+    }
+
+    /**
+     * @test
+     */
+    public function it_ignores_an_update_with_equal_prices_on_uitpas_prices(): void
```
Comment (bertramakers) [inline tests/Event/EventTest.php:688]: The test name says `with_equal_prices_on_uitpas_prices` but the `updatePriceInfo()` call does not include uitpas prices, so they're not equal?

**Review by bertramakers — APPROVED** [review]
Looks good functionally and well tested! Just not 100% sure about the test names

File: tests/Event/EventTest.php (hunk 3 above)
Comment (JonasVHG) [reply tests/Event/EventTest.php]: I'll try to improve the names, I created tests for:
- update on Offer with different Base/Tariffs & No UiTPAS -> expects base & Tariffs update, UiTPAS should stay the same `it_keeps_existing_uitpas_prices_on_price_info_update()`
- update on Offer with different Base/Tariffs & different UitPAS -> expects only Base & Tariffs update, UiTPas should stay the same `it_ignores_an_update_of_uitpas_prices()`
- update an Offer without same Base Tariffs & no UiTPAS -> expects no Event recorded `it_ignores_an_update_with_equal_prices_without_uitpas()`
- update on Offer with same Base/Tariffs & different UiTPAS Prices -> expects no Event recorded `it_ignores_an_update_with_only_different_uitpas_prices()`

Comment (bertramakers) [reply tests/Event/EventTest.php]: Is the third point:

> update an Offer without same Base Tariffs & no UiTPAS -> expects no Event recorded `it_ignores_an_update_with_equal_prices_without_uitpas()`

intended to be:

> update an Offer **with** same Base Tariffs & no UiTPAS -> expects no Event recorded `it_ignores_an_update_with_equal_prices_without_uitpas()`

?

Comment (JonasVHG) [reply tests/Event/EventTest.php]: Indeed, that was a confusing typo.

**Review by bertramakers — APPROVED** [review]
Super clear now!

File: tests/Event/EventTest.php (hunk 3 above)
Comment (bertramakers) [reply tests/Event/EventTest.php]: Alrighty, aside from that it was clear to me! 👍

**Comment by bertramakers** [comment]
Isn't there a composer update needed of the `udb3-json-schemas` package to update the JSON schema for priceInfo btw?

**Comment by JonasVHG** [comment]
> Isn't there a composer update needed of the `cultuurnet/udb3-json-schemas` package to update the JSON schema for priceInfo btw?

Good find, I've added it.

**Review by bertramakers — APPROVED** [review]

**Comment by bertramakers** [comment]
Looks complete to me now!

**Review by LucWollants — APPROVED** [review]
Nice work!

**Pull request merged**